hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/RowCounter.java

```diff
@@ -95,6 +118,36 @@ public static enum Counters {
     public void map(ImmutableBytesWritable row, Result values, Context context) throws IOException {
       // Count every row containing data, whether it's in qualifiers or values
       context.getCounter(Counters.ROWS).increment(1);
+
+      if (countDeleteMarkers) {
+        context.getCounter(Counters.CELLS).increment(values.size());
```

Comment:
We should have another flag for cells, should not be clubbed with flag for delete markers

Reply:
Cell counter will only make sense if we have countDeleteMarkers flag set as we would like to get a comparison of cell with and without delete markers.
Otherwise, we already have a cell counter tool.

Reply:
If I understand correctly we can only have cell count with delete marker, how do you get without case here? 
Also if you want to count only deleted cells, why not do it when we know cell has a delete type?

Reply:
But this is not intuitive as we have no way of counting when delete markers is disabled within the job. If another tool is to be leveraged for counting cells, same should be update to work for deleted rows also

Reply:
>as we have no way of counting when delete markers is disabled within the job.

@NihalJain , I am not sure if I got you. However, the user can always control if he needs the delete markers count or not by the specifying the flag (--countDeleteMarkers) while triggering the job.

Reply:
On second thoughts, I have removed cell counter altogether because it will not make much sense if we have multiple versions for a given column as the scan used in RowCounter is set to show only 1 version. Having it will cause more confusion as to what it's value represents.

Reply:
yes we should add it if needed in cellcounter job.